src/or/dirauth/shared_random.h

```diff
@@ -101,11 +101,40 @@ typedef struct sr_commit_t {
 
 /* API */
 
-/* Public methods: */
+/* Public methods used _outside_ of the module.
+ *
+ * We need to nullify them if the module is disabled. */
+#ifdef HAVE_MODULE_DIRAUTH
 
 int sr_init(int save_to_disk);
 void sr_save_and_cleanup(void);
 void sr_act_post_consensus(const networkstatus_t *consensus);
+
+#else /* HAVE_MODULE_DIRAUTH */
+
+static inline int
+sr_init(int save_to_disk)
+{
+  (void) save_to_disk;
+  /* Always return success. */
+  return 0;
+}
+
+static inline void
+sr_save_and_cleanup(void)
+{
+}
+
+static inline void
+sr_act_post_consensus(const networkstatus_t *consensus)
```

Comment:
I'm not sure that this function is one it makes sense to expose in the case when this module is disabled. Do we really need to do so?  It doesn't seem as logical as the "init" and "cleanup" functions.

Reply:
Indeed, it should be dirauth only. However, it is used in `directory.c` when a consensus is fetched and if we are acting as a dirauth, we call that post consensus function.

It should be made dirauth only when we refactor the directory.c file for all `authdir_mode_v3()` places that should go in `src/or/dirauth/` at that point.

Reply:
Okay.  I'll look for things we can #ifdef postmerge...